src/app/private/scripts/controllers/announcement/createAnnouncementController.js

```diff
@@ -139,36 +128,13 @@ angular.module('playerApp').controller('createAnnouncementCtrl', ['$rootScope',
      * @memberOf Controllers.createAnnouncementCtrl
      * @param {int} [index] [description]
      */
-    createAnn.removeLink = function (index) {
-      createAnn.repeatableWebLinks.splice(index, 1)
-      if (createAnn.data.links) {
-        delete createAnn.data.links[index]
-      }
-      createAnn.showUrlField = !!createAnn.repeatableWebLinks.length
-      createAnn.enableRecepientBtn()
-    }
-
-    /**
-     * @method previewAnn
-     * @desc - preview announcement
-     * @memberOf Controllers.createAnnouncementCtrl
-     */
-    createAnn.previewAnn = function () {
-      createAnn.linkArray = []
-      if (createAnn.data.links) {
-        angular.forEach(createAnn.data.links, function (value, key) {
-          if (value.trim().length) {
-            createAnn.linkArray.push(value)
-          }
-        })
-      }
-      createAnn.previewData = {
-        'details': { 'type': createAnn.data.type, 'title': createAnn.data.title, 'description': createAnn.data.description, 'from': createAnn.data.from },
-        'sourceid': $rootScope.rootOrgId,
-        'links': createAnn.linkArray,
-        'target': ['teachers'],
-        'attachments': createAnn.attachment
+    composeAnn.removeLink = function (index) {
+      composeAnn.repeatableWebLinks.splice(index, 1)
```

Comment:
this state change should be made on the AnnouncementModel class not on the controller state.